modules/gdscript/gdscript_analyzer.cpp

```diff
@@ -4306,7 +4306,12 @@ GDScriptParser::DataType GDScriptAnalyzer::type_from_variant(const Variant &p_va
 	result.builtin_type = p_value.get_type();
 	result.type_source = GDScriptParser::DataType::ANNOTATED_EXPLICIT; // Constant has explicit type.
 
-	if (p_value.get_type() == Variant::OBJECT) {
+	if (p_value.get_type() == Variant::ARRAY && p_source->type == GDScriptParser::Node::BINARY_OPERATOR) {
```

Comment:
Ideally `p_source` should not matter, it is for debugging purposes (warnings and errors).

To get an element type combo of `type_from_metatype` and relevant `make_builtin_meta_type`/`make_native_meta_type`/`make_script_meta_type` can be used.

Reply:
Thanks for the help, it looks much cleaner now!

Reply:
`type_from_metatype` calls on args for `set_container_element_type` are missing.

Reply:
Oops, missed that, it's there now!